kong/api/crud_helpers.lua

```diff
@@ -128,23 +138,38 @@ function _M.paginated_set(self, dao_collection)
 
   local next_url
   if offset then
-    offset = ngx.encode_base64(offset)
+    offset = encode_base64(offset)
     next_url = self:build_url(self.req.parsed_url.path, {
-      port = self.req.parsed_url.port,
-      query = ngx.encode_args {
+      port     = self.req.parsed_url.port,
+      query    = encode_args {
         offset = offset,
-        size = size
+        size   = size
       }
     })
   end
 
-  return responses.send_HTTP_OK {
+  local data
+
+  if #rows == 0 then
     -- FIXME: remove and stick to previous `empty_array_mt` metatable
     -- assignment once https://github.com/openresty/lua-cjson/pull/16
     -- is included in the OpenResty release we use.
-    data = #rows > 0 and rows or cjson.empty_array,
-    total = total_count,
-    offset = offset,
+    data = cjson.empty_array
+
+  else
+    data = rows
+
+    if type(cb) == "function" then
+      for _, row in ipairs(rows) do
+        cb(row)
+      end
```

Comment:
well, I thought about it, but I left that door open for the future as I didn't need that. In general I think returning `table` (a replacement row when you really want to replace the row instead of modifying the reference table) is fine. I don't agree that with `nil` you should remove the row. Maybe `false` could do that. Because then you lose simple reference modification and always require return value. Removing rows in a paged set is problematic as it would change the page size. That means that removing rows should happen before paging.